### DISCUSSION TITLE
How to package and build the localization of Chinese Simplified Chinese

### DISCUSSION THREAD

**Discussion by VolareYang** [dbeaver/dbeaver#36556] [DBeaver Development] [answered]
System information:
Operating system (distribution) and version: Windows 11 Enterprise Edition (Simplified Chinese),23H2
DBeaver version: 24.2.4
Additional extensions

Hello!
I am a dbeaver community enthusiast and I would like to contribute to the localization of DBeaver (Chinese Simplified). When I clone the dbeaver repository, compile and build locally, execute build.cmd packaging, go to path: dbeaver\product\community\target\products\org.jkiss.dbeaver.core.product\win32\win32\x86_64\dbeaver\dbeaver.exe  and executes it, it is found that there are two problems, and the 
1.dbeaver.exe icon is changed to the eclipse icon.

2. Execution dbeaver.exe found that some Chinese were not translated, the same problem as @bianyun1981 engineers(https://github.com/orgs/dbeaver/discussions/14914), and then I also made some questions in the next main, how to automatically integrate the Babel package when packaging.
![Image20241216114001671](https://github.com/user-attachments/assets/41fc9410-5642-4a50-be75-de6726df0d79)

With all that said, I need some help with the issue of packaging into .exe executable and the Babel language pack. Is there a how-to procedure and a guide to read, thank you!

**Comment by EastLord** [comment, marked as answer]
https://github.com/dbeaver/dbeaver/wiki/Localization
https://github.com/dbeaver/dbeaver/wiki/Develop-in-IDEA
https://www.jetbrains.com/help/idea/internationalization-and-localization.html
![image](https://github.com/user-attachments/assets/82dd0312-853f-4d6c-bf88-d584de0f57ca)

**Reply by VolareYang** [reply]
Thank you! @EastLord 
Later, the Chinese Simplified Chinese localization deployment can slowly add the untranslated part of the development of the dbeaver source code. But I use the .exe package downloaded from the dbeaver official website, switch Chinese Simplified open it and find that it is basically translated, but I compile and package it locally into a .exe package and open it and find that some of it is not translated, such as the menu bar Search, like the one shown in the following picture. Then how to solve it, I need your help, thanks!
![image](https://github.com/user-attachments/assets/8cfede8c-7be2-4286-b259-3dc8c57d028b)

**Reply by EastLord** [reply]
org.jkiss.dbeaver.ui.search  Module

**Reply by VolareYang** [reply]
Hello!@EastLord
This module is there, and there is a Chinese translation corresponding to Search, but it doesn't seem to work, what is the reason, need your help, thank you!

![image](https://github.com/user-attachments/assets/7c98416c-4417-45da-ba37-aa547ebf6ef9)
